src/core/init/directfile_content_initializer.ts

```diff
@@ -253,15 +254,30 @@ function getDirectFileInitialTime(
   }
 
   const duration = mediaElement.duration;
-  if (duration == null || !isFinite(duration)) {
-    log.warn("startAt.fromLastPosition set but no known duration, " +
-             "beginning at 0.");
-    return 0;
-  }
 
   if (typeof startAt.fromLastPosition === "number") {
+    if (isNullOrUndefined(duration) || !isFinite(duration)) {
+      log.warn("startAt.fromLastPosition set but no known duration, " +
+               "beginning at 0.");
+      return 0;
+    }
     return Math.max(0, duration + startAt.fromLastPosition);
+  } else if (typeof startAt.fromLivePosition === "number") {
+    const livePosition = mediaElement.seekable.length > 0 ?
+      mediaElement.seekable.end(0) :
+      duration;
+    if (isNullOrUndefined(livePosition)) {
+      log.warn("startAt.fromLivePosition set but no known live position, " +
+               "beginning at 0.");
+      return 0;
+    }
+    return Math.max(0, livePosition + startAt.fromLivePosition);
```

Comment:
In my mind, it could still be theoretically possible to begin after the live position if there's already content available for it.

It may be surprising, but the fact that there might be content already available after the live position is what started the need for this development.

Reply:
OK, that might be a very specific case but I understand your point

Reply:
OK, that might be a very specific case but I understand your point